libbeat/publisher/queue/diskqueue/compression_test.go

```diff
@@ -0,0 +1,157 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package diskqueue
+
+import (
+	"bytes"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type nopWriteCloser struct {
+	io.Writer
+}
+
+func NopWriteCloser(w io.Writer) io.WriteCloser {
+	return nopWriteCloser{w}
+}
+func (nopWriteCloser) Close() error { return nil }
+
+func TestCompressionReader(t *testing.T) {
+	tests := map[string]struct {
+		plaintext  []byte
+		compressed []byte
+	}{
+		"abc 1.9.3 lz4": {
+			plaintext: []byte("abc"),
+			compressed: []byte{
+				0x04, 0x22, 0x4d, 0x18,
+				0x64, 0x40, 0xa7, 0x04,
+				0x00, 0x00, 0x80, 0x61,
+				0x62, 0x63, 0x0a, 0x00,
+				0x00, 0x00, 0x00, 0x6c,
+				0x3e, 0x7b, 0x08, 0x00},
+		},
+		"abc pierrec lz4": {
```

Comment:
What is the difference between the two test cases here?

Reply:
one was written with C library, other with golang library.  I updated the names to reflect this.